Sources/ProjectDescription/Config.swift

```diff
@@ -124,8 +137,15 @@ extension Config.GenerationOptions {
             try container.decode(Bool.self, forKey: .disableShowEnvironmentVarsInScriptPhases)
         {
             self = .disableShowEnvironmentVarsInScriptPhases
-        } else if container.allKeys.contains(.enableCodeCoverage), try container.decode(Bool.self, forKey: .enableCodeCoverage) {
-            self = .enableCodeCoverage
+        } else if container.allKeys.contains(.enableCodeCoverage) {
+            do {
+                // try to maintain backwards compatibility
+                _ = try container.decode(Bool.self, forKey: .enableCodeCoverage)
+                self = .enableCodeCoverage(.all)
+            } catch {
+                let mode = try container.decode(CodeCoverageMode.self, forKey: .enableCodeCoverage)
+                self = .enableCodeCoverage(mode)
+            }
```

Comment:
Why do you need the backward compatibility? Is there a case in which you would load a serialized Config generated with an old tuist version?

Reply:
I don't think so, but I don't see all possible scenarios and I think it is better to play it safe than introduce some ungly, unexpected issues.

Reply:
The backward compatibility has already been achieved with [this change](https://github.com/tuist/tuist/pull/3267/files#diff-f3cd2eee98f21da8a7a3507534230676f207abd792e2af41bf13e879edca28a2R41). Therefore we can remove some code from here.